include/bio/map_io/sam_tag_dictionary.hpp

```diff
@@ -0,0 +1,746 @@
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides the bio::map_io::sam_tag_dictionary class and auxiliaries.
+ * \author Svenja Mehringer <svenja.mehringer AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <map>
+#include <seqan3/std/charconv>
+#include <seqan3/std/concepts>
+#include <variant>
+
+#include <seqan3/core/detail/template_inspection.hpp>
+#include <seqan3/utility/char_operations/predicate.hpp>
+#include <seqan3/utility/concept/exposition_only/core_language.hpp>
+
+#include <bio/detail/views_eager_split.hpp>
+
+namespace bio::detail
+{
+//!\brief std::variant of allowed types for optional tag fields of the SAM format.
+//!\ingroup io_sam_file
+using sam_tag_variant = std::variant<char,
+                                     int32_t,
+                                     float,
+                                     std::string,
+                                     std::vector<std::byte>,
+                                     std::vector<int8_t>,
+                                     std::vector<uint8_t>,
+                                     std::vector<int16_t>,
+                                     std::vector<uint16_t>,
+                                     std::vector<int32_t>,
+                                     std::vector<uint32_t>,
+                                     std::vector<float>>;
+
+//!\brief Each SAM tag type char identifier. Index corresponds to the bio::detail::sam_tag_variant types.
+//!\ingroup io_sam_file
+char constexpr sam_tag_type_char[12]       = {'A', 'i', 'f', 'Z', 'H', 'B', 'B', 'B', 'B', 'B', 'B', 'B'};
+//!\brief Each types SAM tag type extra char id. Index corresponds to the bio::detail::sam_tag_variant types.
+//!\ingroup io_sam_file
+char constexpr sam_tag_type_char_extra[12] = {'\0', '\0', '\0', '\0', '\0', 'c', 'C', 's', 'S', 'i', 'I', 'f'};
+} // namespace bio::detail
+
+namespace bio
+{
+
+inline namespace literals
+{
+
+/*!\name Other literals
+ * \{
+ */
+/*!\brief The SAM tag literal, such that tags can be used in constant expressions.
+ * \ingroup io_sam_file
+ * \tparam char_t The char type. Usually `char`. Parameter pack `...s` must be of length 2 since SAM tags consist of two
+ *                letters (char0 and char1).
+ * \relatesalso bio::map_io::sam_tag_dictionary
+ * \returns The unique identifier of the SAM tag computed by char0 * 128 + char1.
+ *
+ * \details
+ *
+ * A SAM tag consists of two letters, initialized via the string literal ""_tag, which delegate to its unique id.
+ *
+ * \snippet test/snippet/io/sam_file/sam_tag_dictionary/sam_tag_dictionary.cpp tag
+ *
+ * The purpose of those tags is to fill or query the bio::map_io::sam_tag_dictionary for a specific key (tag_id) and
+ * retrieve the corresponding value.
+ */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
+template <typename char_t, char_t... s>
+constexpr uint16_t operator""_tag()
```

Comment:
I would prefer if we can replace this a `bio::vtag`, but this doesn't have to happen now. I will have a think on it.

Reply:
Alright, are you keeping track of this?

Reply:
I have now started adding Github discussion items for these things. Feel free to add information there.